sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/FileConfiguration.java

```diff
@@ -86,10 +86,21 @@ public static OpenTelemetrySdk parseAndCreate(InputStream inputStream) {
    * @throws ConfigurationException if unable to interpret
    */
   public static OpenTelemetrySdk create(OpenTelemetryConfiguration configurationModel) {
+    return create(configurationModel, SpiHelper.create(FileConfiguration.class.getClassLoader()));
+  }
+
+  /**
+   * Interpret the {@code configurationModel} to create {@link OpenTelemetrySdk} instance
+   * corresponding to the configuration.
+   *
+   * @param configurationModel the configuration model
+   * @return the {@link OpenTelemetrySdk}
+   * @throws ConfigurationException if unable to interpret
+   */
+  public static OpenTelemetrySdk create(
+      OpenTelemetryConfiguration configurationModel, SpiHelper spiHelper) {
```

Comment:
`public static io.opentelemetry.sdk.trace.samplers.Sampler createSampler(StructuredConfigProperties genericSamplerModel)` method also needs to load `ComponentProvider`s using the agent-specified class loader. Since this is a helper method which we expect `ComponentProviders` to call, we need to find a way to embed the `ClassLoader` / `SpiHelper` in the `StructuredConfigProperties` such that:
- Agent loads SDK via declarative configuration
- ComponentProviders are loaded via the agent-specified class loader
- Any ComponentProvider which uses these utility functions to load delegate components (i.e. a Sampler loads and delegates to another sampler) has ComponentProviders using the same agent-specified class loader

Reply:
TODO: solve this before merging